_projects/go_bears_on_bsd.md

```diff
@@ -0,0 +1,57 @@
+---
+collaborating_projects:
+  - coala
+desc: "Explore improvements to coala's dependency management through having Go Linter installed in *BSD system automatially."
+difficulty: medium
+initiatives:
+  - GSoC
+  - RGSoC
+issues:
+  - "https://gitlab.com/coala/package_manager/issues/149"
+  - "https://github.com/coala/coala-bears/issues/2481"
+markdown: go_bears_on_bsd.md
+mentors:
+  - xferra
+  - yukiisbored
+name: "Linter Dependency Management from a Bare-Metal System"
+requirements:
+  - "At least one patch to the coala core should be accepted and merged."
+  - "The proposal needs to show familiarity with coala's dependency management implementation."
+status: in_progress
+tags:
+  - Automation
+  - Installation
+  - Dependency
+---
+
+`coala-bears` are of two types, Native Bears which implement all the business logic themselves and Linter Bears which wrap existing linters and checkers. These Linter Bears have dependencies which need to be installed before they can be executed.
+
+`dependency_management` provides Python APIs for dependency management on various languages and different systems. It is used extensively by coala, however, it lacks a lot of the key implementation and documentation in its current state.
+
+The student will implement features to coala’s automated management system. `dependency_management` should be able to manage language runtimes and operate language-specific package managers for dependencies. It should also resolve package names and versions automatically on multiple operating systems.
+
+The end goal is to be able to bootstrap and recursively install all dependencies on any systems without anything installed other than a base system. The scope of a complete solution is too large. The student made a proposal with suitable scope, which is to automate the installation of a Go Linter Bear onto bare-metal BSDs. The project may also provide a blueprint for future implementation and enhancement.
```

Comment:
`The student` isnt suitable for a project intro.

Just describe your project here.

For example, what are all the current go linters in coala-bears?

And what are the typical problems with Go dependencies?  Especially how does Go do versioning?

e.g. https://github.com/coala/coala-bears/issues/2457
See also https://github.com/coala/coala-bears/issues/1472 and https://github.com/coala/coala-bears/issues/1488 and other problems we have encountered.
And https://github.com/coala/coala/issues/2752#issuecomment-382411052 for Go problems which are even older and probably never solved and likely are now unsolvable.

As you can see, Go is ok with breaking old tools, and de-supports old versions quite quickly.

In the milestones `The student ...` below is correct.